src/operator/tensor/matrix_op.cc

```diff
@@ -354,6 +354,60 @@ NNVM_REGISTER_OP(dot)
 
     dot(x,y)[i,j,a,b] = sum(x[i,j,:]*y[:,a,b])
 
+  Example::
+
+    x = [[[  0.,   1.,   2.,   3.],
+	  [  4.,   5.,   6.,   7.],
+	  [  8.,   9.,  10.,  11.]],
+
+	 [[ 12.,  13.,  14.,  15.],
```

Comment:
Can you elaborate more on your suggestion? I have tried to demonstrate that `dot(x,y)[i,j,k,l] = sum(x[i,j,:]*y[:,k,l])` through the example, similar to the numpy docs. Do you want me to show how array is computed using arange and reshape instead of writing actual elements of the array? I can change the example to make it 2by2by2 by 2by2by2 and use  smaller numbers.

Please let me know which of the following two options you prefer ?

Option1::


          x = [[[ 0.,   1.],
                [ 1.,   0.],

               [[ 1.,   0.],
                [ 0.,   1.]]]

          y = [[[ 0.,   1.],
                [ 2.,   3.],

               [[ 4.,   5.],
                [ 6.,   7.]]]

          dot(x,y) = [[[[ 0.,   1.],
                        [ 2.,   3.]],

                       [[ 4.,   5.],
                        [ 6.,   7.]]],

                      [[[ 0.,   1.],
                        [ 2.,   3.]],

                       [[ 4.,   5.],
                        [ 6.,   7.]]]]

          dot(x,y)[1,1,1,1] = 7.0

          sum(x[1,1,:]*y[:,1,1]) = 7.0

Option 2::

          x = reshape([0,1,2,3,4,5,6,7], shape=(2,2,2))
          y = reshape([7,6,5,4,3,2,1,0], shape=(2,2,2))
          dot(x,y)[1,1,1,1] = 24
          sum(x[1,1,:]*y[:,1,1]) = 24